routes/controllers/snippets.controllers.js

```diff
@@ -0,0 +1,131 @@
+const mongoose = require("mongoose");
+const createError = require("http-errors");
+const jwt = require("jsonwebtoken");
+
+const Snippet = require("../../models/Snippet");
+const Hashtag = require("../../models/Hashtag");
+
+const {
+  INVALID_ID,
+  INVALID_REQUEST,
+  NO_AUTHORITY_TO_ACCESS,
+  NOT_FOUND,
+  UNEXPECTED_ERROR,
+  OK,
+} = require("../../constants/messages");
+
+const getSnippetList = async (req, res, next) => {
+  const { language, search } = req.query;
+
+  const hashtagList = search?.split(" ");
+
+  try {
+    const targets = {};
+
+    language && (targets.language = language);
+    hashtagList && (targets.hashtagList = { $all: hashtagList });
+
+    const snippetList = await Snippet.find(targets);
+
+    res
+      .status(200)
+      .send({ result: OK, snippetList });
+  } catch (error) {
+    next(error);
+  }
+};
+
+const getSnippet = async (req, res, next) => {
+  const { id } = req.params;
+
+  try {
+    const snippet = await Snippet
+      .findById(id)
+      .populate(["creator", "poster", "commentList.creator"]);
+
+    const isInvalid = snippet === null;
+
+    if (isInvalid) {
+      throw createError(404, INVALID_ID);
+    }
+
+    res.send({ result: OK, snippet });
+  } catch (error) {
+    if (error.status) {
+      next(error);
+      return;
+    }
+
+    if (error instanceof mongoose.Error.CastError) {
+      next(createError(422, INVALID_REQUEST));
+      return;
```

Comment:
마찬가지로 이 부분도 개행 추가 부탁드려요!

```suggestion
      next(createError(422, INVALID_REQUEST));
      
      return;
```